app/src/main/java/com/omkarmoghe/pokemap/views/map/MapWrapperFragment.java

```diff
@@ -505,14 +510,21 @@ public void onFetch(Bitmap bitmap) {
 
                                     //Setting marker since we got image
                                     //int resourceID = getResources().getIdentifier("p" + poke.getPokemonId().getNumber(), "drawable", getActivity().getPackageName());
-                                    Marker marker = mGoogleMap.addMarker(new MarkerOptions()
+                                    final Marker marker = mGoogleMap.addMarker(new MarkerOptions()
                                             .position(new LatLng(poke.getLatitude(), poke.getLongitude()))
                                             .title(PokemonIdUtils.getLocalePokemonName(getContext(), poke.getPokemonId().name()))
                                             .icon(bitmapDescriptor)
                                             .zIndex(MapHelper.LAYER_POKEMONS)
                                             .anchor(0.5f, 0.5f));
                                     //adding pokemons to list to be removed on next search
-                                    markerList.put(poke.getSpawnPointId(), new PokemonMarkerExtended(poke, marker));
+                                    getActivity().runOnUiThread(new Runnable() {
```

Comment:
Had to add that as sometimes for some reason it wasn't running on the UI thread which caused the messages to not be posted.